orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/HealthHelper.groovy

```diff
@@ -112,4 +112,38 @@ class HealthHelper {
     boolean noneAreDown = !healths.any { Map health -> health.state == 'Down' }
     return someAreUp && noneAreDown
   }
+
+  static HealthCountSnapshot getHealthCountSnapshot(List<Map> instances, Collection<String> interestingHealthProviderNames) {
+    HealthCountSnapshot snapshot = new HealthCountSnapshot()
+    instances.each { instance ->
+      List<Map> healths = filterHealths(instance, interestingHealthProviderNames)
+      if (areSomeUpConsideringPlatformHealth(healths, interestingHealthProviderNames, healths.any { Map health -> health.state == 'Up' } )) {
+        snapshot.up++
+      } else if (isDownConsideringPlatformHealth(healths) ) {
+        snapshot.down++
+      } else if (healths.any { it.state == 'OutOfService' } ) {
+        snapshot.outOfService++
+      } else if (healths.any { it.state == 'Starting' } ) {
+        snapshot.starting++
+      } else if (healths.every { it.state == 'Succeeded' } ) {
+        snapshot.succeeded++
+      } else if (healths.any { it.state == 'Failed' } ) {
+        snapshot.failed++
+      } else {
+        snapshot.unknown++
+      }
+    }
+    snapshot
+  }
+
+  static class HealthCountSnapshot {
+    int up
+    int down
+    int unknown
```

Comment:
If `unknown` were defined last, the order would match the order they are used in the `if..else-if`.